### PR TITLE
Download jdtls to extension work directory if not exist

### PR DESCRIPTION
**Pull request by azam**
Hi!
This change downloads jdtls if it does not exist on $PATH.
I couldn't find any CLA nor guidelines for zed-extensions, so do please advise if there is any.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -2,14 +2,165 @@ use zed_extension_api::{
...
+                    language_server_id,
+                    &zed::LanguageServerInstallationStatus::Failed(
+                        "no version specified in settings and no latest release found".to_string(),
+                    ),
+                );
+                return Err(
+                    "no version specified in settings and no latest release found".to_string(),
+                );
+            }
+        };
```
Comment (valentinegb) [inline src/lib.rs:71]: Do correct me if I'm wrong, but couldn't this match statement be (off the top of my head) something like `.or_else(|| /* ... */).ok_or("Error message")?` instead?

(Thanks so much for this btw! This has absolutely been needed ^^)

**Review by valentinegb — COMMENTED** [review]
Yeah, besides that one thing, this looks fantastic! I'll test this lightly when I get the chance (I'm assuming you've already done a good amount of testing) but that might be later in the day, sorry

File: src/lib.rs (hunk 1 above)
Comment (azam) [reply src/lib.rs]: @valentinegb 
Yeah, this is some cruft when I tried to use `zed::GithubReleaseOption` to fetch latest versions, but realized they only add tags, not releases.
I simplified it to default to fixed version. -> 65b3bccfffb5e23e06835a9d33456c415401d92f

**Comment by notpeter** [comment]
@cla-bot check

**Pull request merged**